File: typespec/problem/admin/totp.ts
import type { Details } from "../details.ts";

export const TOTPAlreadyEnabledError: Readonly<Details> = {
  type: "vetchium-problem-details/totp-already-enabled",
  title: "TOTP already enabled",
  status: 409,
  detail: "The admin user already has confirmed TOTP",
};

export const InvalidTOTPEnrollmentError: Readonly<Details> = {
  type: "vetchium-problem-details/invalid-totp-enrollment",
  title: "Invalid TOTP enrollment",
  status: 409,
  detail:
    "TOTP enrollment is invalid, expired, consumed, or belongs to another user",
};

export const IncorrectRecoveryCodeError: Readonly<Details> = {
  type: "vetchium-problem-details/incorrect-recovery-code",
  title: "Incorrect recovery code",
  status: 422,
  detail: "The recovery code is incorrect, consumed, or unavailable",
};

export const TOTPNotEnabledError: Readonly<Details> = {
  type: "vetchium-problem-details/totp-not-enabled",
  title: "TOTP not enabled",
  status: 409,
  detail: "The admin user does not have confirmed TOTP",
};

File: typespec/problem/admin/totp.go
package admin

import "github.com/vetchium/src/typespec/problem"

var TOTPAlreadyEnabledError = problem.Details{
	Type:   "vetchium-problem-details/totp-already-enabled",
	Title:  "TOTP already enabled",
	Status: 409,
	Detail: "The admin user already has confirmed TOTP",
}

var InvalidTOTPEnrollmentError = problem.Details{
	Type:   "vetchium-problem-details/invalid-totp-enrollment",
	Title:  "Invalid TOTP enrollment",
	Status: 409,
	Detail: "TOTP enrollment is invalid, expired, consumed, or belongs to " +
		"another user",
}

var IncorrectRecoveryCodeError = problem.Details{
	Type:   "vetchium-problem-details/incorrect-recovery-code",
	Title:  "Incorrect recovery code",
	Status: 422,
	Detail: "The recovery code is incorrect, consumed, or unavailable",
}

var TOTPNotEnabledError = problem.Details{
	Type:   "vetchium-problem-details/totp-not-enabled",
	Title:  "TOTP not enabled",
	Status: 409,
	Detail: "The admin user does not have confirmed TOTP",
}
